dotCMS/src/main/java/com/dotcms/contenttype/business/RelationshipFactoryImpl.java

```diff
@@ -401,44 +426,88 @@ private void delete(Relationship relationship, Boolean keepTreeRecords) throws D
-
-            HibernateUtil dh = new HibernateUtil(com.dotmarketing.portlets.contentlet.business.Contentlet.class);
+    private void deleteInodeInDB(final String inode) throws DotDataException{
+        DotConnect dc = new DotConnect();
+        dc.setSQL(sql.DELETE_INODE);
+        dc.addParam(inode);
+        dc.loadResult();
+    }
 
-            String sql = "SELECT {contentlet.*} from contentlet contentlet, inode contentlet_1_, contentlet_version_info vi, tree tree1 "
-            		+ "where tree1.parent = ? and tree1.relation_type = ?  "
-                    + "and tree1.child = contentlet.identifier "
-                    + "and contentlet.inode = contentlet_1_.inode and vi.identifier=contentlet.identifier "
-                    + "and " + ((live)?"vi.live_inode":"vi.working_inode") + " = contentlet.inode ";
+	public  List<Contentlet> dbRelatedContentByParent(final String parentInode, final String relationType, final boolean live,
+            final String orderBy) throws DotDataException {
+
+	    StringBuilder query = new StringBuilder("select cont1.inode, show_on_menu, title, mod_date, mod_user, sort_order, friendly_name, structure_inode, last_review, next_review, "
+                + "review_interval, disabled_wysiwyg, cont1.identifier, language_id, date1, date2, date3, date4, date5, date6, date7, date8, date9, date10, "
+                + "date11, date12, date13, date14, date15, date16, date17, date18, date19, date20, date21, date22, date23, date24, date25, text1, text2, "
+                + "text3, text4, text5, text6, text7, text8, text9, text10, text11, text12, text13, text14, text15, text16, text17, text18, text19, "
+                + "text20, text21, text22, text23, text24, text25, text_area1, text_area2, text_area3, text_area4, text_area5, text_area6, text_area7, "
+                + "text_area8, text_area9, text_area10, text_area11, text_area12, text_area13, text_area14, text_area15, text_area16, text_area17, "
+                + "text_area18, text_area19, text_area20, text_area21, text_area22, text_area23, text_area24, text_area25, integer1, integer2, integer3, "
+                + "integer4, integer5, integer6, integer7, integer8, integer9, integer10, integer11, integer12, integer13, integer14, integer15, "
+                + "integer16, integer17, integer18, integer19, integer20, integer21, integer22, integer23, integer24, integer25, \"float1\", "
+                + "\"float2\", \"float3\", \"float4\", \"float5\", \"float6\", \"float7\", \"float8\", \"float9\", \"float10\", \"float11\", \"float12\", "
+                + "\"float13\", \"float14\", \"float15\", \"float16\", \"float17\", \"float18\", \"float19\", \"float20\", \"float21\", \"float22\", "
+                + "\"float23\", \"float24\", \"float25\", bool1, bool2, bool3, bool4, bool5, bool6, bool7, bool8, bool9, bool10, bool11, bool12, bool13, "
+                + "bool14, bool15, bool16, bool17, bool18, bool19, bool20, bool21, bool22, bool23, bool24, bool25, ")
```

Comment:
do we need to name all the parameters in the `select`?

Reply:
yes, from what I saw in other queries we need to name every parameter